pkg/highlight/highlighter.go

```diff
@@ -186,7 +186,9 @@ func (h *Highlighter) highlightRegions(fullHighlights []Group, start int, lineNu
 							h.lastEnd = start + endMatches[endIdx][1]
```

Comment:
What about this remaining check in `storePatterns()`:

```
		if curRegion == nil || curRegion.group == curRegion.limitGroup || p.group == curRegion.limitGroup {
```

Looks like it doesn't make sense? `storePatterns()` is about patterns inside a region, i.e. between the region's `limitGroup`-matching patterns, so `storePatterns()` itself shouldn't have anything to do with `limitGroup`?

Reply:
Indeed, looks like historical takeover. This makes the whole if condition superfluous.
I removed it locally and opened all my test files again. There was no obvious fault.